### PR TITLE
[LETS-639] Refactor async_disconnect_handler to use it in tran_server too

### PR DESCRIPTION
**Pull request by wrlawodms**
http://jira.cubrid.org/browse/LETS-639

We need to add the "disconnection request" from PS to TS. In the request handler, it will try to delete connection_handler object, but it ends up kind of deadlock. The request handler is called by the internal request receiver thread in connection_handler. If it tries to delete itself, it tries to join itself in the dtor (~connection_handler). To address this issue, we can pass the deletion job to another thread. At first, I come up with using a detached thread, but the `async_disconnect_handler` has been introduced seemingly for this purpose in #3407 and a more robust way. So, I make it a template class to be used in not only page_server but also tran_server.

- No functional change
- Make the async_disconnect_handler a template class to be used in common for tran_server and page_server.
- An prior issue for the main PR for LETS-639.

### CODE REVIEW

File: src/server/async_disconnect_handler.hpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2008 Search Solution Corporation
```
Comment (joohok) [inline src/server/async_disconnect_handler.hpp]: I think we don't need this line
```suggestion

```

**Review by joohok — APPROVED** [review]

**Review by hornetmj — APPROVED** [review]

File: src/server/async_disconnect_handler.cpp [hunk 2]
```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2008 Search Solution Corporation
```
Comment (wrlawodms) [inline src/server/async_disconnect_handler.cpp]:
```suggestion
```

File: src/server/async_disconnect_handler.cpp [hunk 3]
```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2016 CUBRID Corporation
```
Comment (joohok) [inline src/server/async_disconnect_handler.cpp]:
```suggestion
  *
  * Copyright 2016 CUBRID Corporation
```

Sorry for wrong suggestion...
I followed the license in schema_class_truncator.hpp

File: src/server/async_disconnect_handler.hpp [hunk 4]
```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2016 CUBRID Corporation
```
Comment (wrlawodms) [inline src/server/async_disconnect_handler.hpp:3]:
```suggestion
 *
 * Copyright 2016 CUBRID Corporation
```

File: src/server/async_disconnect_handler.cpp (hunk 3 above)
Comment (wrlawodms) [reply src/server/async_disconnect_handler.cpp]: I didn't check it too. Thank you. :)

File: src/server/async_disconnect_handler.hpp [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+#include <mutex>
+#include <atomic>
+#include <queue>
+#include <condition_variable>
+
+/* helper class with the task of destroying connnection handlers and, by this,
+ * also waiting for the receive and transmit threads inside the handlers to terminate
+ */
+template <typename T_CONN_HANDLER>
+class async_disconnect_handler
```
Comment (cristiarg) [inline src/server/async_disconnect_handler.hpp:33]: A TODO here would be needed. Or a separate Jira issue.
This class also needs to have a timeout functionality. Based on a system parameter, the class must wait up to a certain time interval to attempt to delete an object. If the delete is not fulfilled in that interval - being that the actual delete of an object depents of finishing some I/O rountrips to the peer server - it is a catastrophic error. (something that works similar to `pgbuf_timed_sleep`)
What do you think?

**Review by cristiarg — APPROVED** [review]

File: src/server/async_disconnect_handler.hpp (hunk 5 above)
Comment (cristiarg) [reply src/server/async_disconnect_handler.hpp:33]: I have added a new epic - http://jira.cubrid.org/browse/LETS-655 - to track this issue. It is not urgent though.

**Pull request merged**